pkg/controllers/managementuser/helm/controller.go

```diff
@@ -336,6 +338,32 @@ func (l *Lifecycle) Run(obj *v3.App, template string, tempDirs *hCommon.HelmPath
 	return l.createAppRevision(obj, template, notes, false)
 }
 
+// pruneOldRevisions checks if the total number of app revisions will not exceed the set maximum.
+// If it will, the method deletes the oldest revision(s).
+func (l *Lifecycle) pruneOldRevisions(obj *v3.App) {
+	if obj.Spec.MaxRevisionCount < 1 {
+		obj.Spec.MaxRevisionCount = defaultMaxRevisionCount
+	}
+	_, projectName := ref.Parse(obj.Spec.ProjectName)
+	appRevisionClient := l.AppRevisionGetter.AppRevisions(projectName)
+	revisions, err := appRevisionClient.List(metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", appLabel, obj.Name),
+	})
+	if err != nil {
+		logrus.Tracef("[helm-controller] Failed to list app revisions for app %s: %v", projectName, err)
+	}
```

Comment:
We should probably return here. Maybe we return an error and the parent function chooses to ignore it?

Reply:
A fair point! Yes, I originally did return it, but now just log it for a few reasons.
1. We should handle it in the method's deferred call, which is hard to do cleanly.
2. This error is not likely to happen on its own, without other `appRevisionClient`'s operations not failing too in adjacent controller handlers and bringing attention to itself.
3. This pruning is not critical enough to stop whatever the controller is doing.

But I completely agree with you, although making the parent ignore the error won't really change anything.

Reply:
Agreed with your points. Regardless, we shouldn't continue the processing here right because those revisions will likely be `nil`, right?

Reply:
Yes, good catch!